### PR TITLE
Add subject, issuer and issuanceDate to the header of credential pages and issuer to list items on home pages

### PR DESCRIPTION
**Pull request by 7opf**
- Closes https://github.com/ZebraDevs/Zebra-Iota-Edge-SDK/issues/26
- Moves init call to main.ts (new changes causes cyclical import in holder app)
- Adds a new component reusable CredentialHeader to each app 
- Removes redundant code in Holder app (**note**: impacts structure of data in LocalStorage - not compatible with previous persistence format, please clear LocalStorage)

### CODE REVIEW

**Comment by 7opf** [comment]
**Device ID**
device id

**Holder**
holder

File: identity-enabler/deviceId-mobile-app/src/components/CredentialHeader.svelte [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    <h2>{credential.type[1]}</h2>
+    {#if !hideDetails}
+        <p>
+            <span>Issued to </span>
+            <a href="{IOTA_IDENTITY_RESOLVER}/{credential.credentialSubject.id}" class="emphasis" target="_blank">
+                {shortenDID(credential.credentialSubject.id)}
+            </a>
+            <span> by </span>
+            <a href="{IOTA_IDENTITY_RESOLVER}/{credential.issuer.id}" class="emphasis" target="_blank"
+                >{credential.issuer.name}
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/components/CredentialHeader.svelte:21]: be careful don't assume issuer.name is going to be always defined ...

File: identity-enabler/deviceId-mobile-app/src/components/CredentialHeader.svelte [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            </a>
+            <span> at </span>
+            <span class="emphasis">{new Date(credential.issuanceDate).toLocaleString()}</span>
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/components/CredentialHeader.svelte:24]: probably you need to be more explicit, specifying a locale, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString

File: identity-enabler/deviceId-mobile-app/rollup.config.js [hunk 3]
```diff
@@ -14,27 +14,6 @@ import * as path from "path";
...
-                stdio: ["ignore", "inherit", "inherit"],
-                shell: true
-            });
-
-            process.on("SIGTERM", toExit);
-            process.on("exit", toExit);
-        }
-    };
-}
-
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/rollup.config.js:37]: why is this change needed in this PR?

File: identity-enabler/deviceId-mobile-app/src/pages/Home.svelte [hunk 4]
```diff
@@ -92,13 +92,13 @@
...
+            {#each localCredentials as vp}
                 <div class="list">
                     <ListItem
                         icon="chip"
-                        onClick={() => navigate("credential", { state: { credential } })}
-                        heading={"IOTA"}
-                        subheading={credential.verifiableCredential.type[1]}
+                        onClick={() => navigate("/credential", { state: { vp } })}
+                        heading={vp.verifiableCredential.type[1]}
+                        subheading="Issued by {vp.verifiableCredential.issuer.name}"
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/deviceId-mobile-app/src/pages/Home.svelte:101]: same comment

File: identity-enabler/holder-mobile-app/rollup.config.js [hunk 5]
```diff
@@ -14,27 +14,6 @@ import * as path from "path";
...
-                stdio: ["ignore", "inherit", "inherit"],
-                shell: true
-            });
-
-            process.on("SIGTERM", toExit);
-            process.on("exit", toExit);
-        }
-    };
-}
-
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/rollup.config.js:37]: same comment

File: identity-enabler/holder-mobile-app/src/components/CredentialHeader.svelte [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    <h2>{credential.type[1]}</h2>
+    {#if !hideDetails}
+        <p>
+            <span>Issued to </span>
+            <a href="{IOTA_IDENTITY_RESOLVER}/{credential.credentialSubject.id}" class="emphasis" target="_blank">
+                {shortenDID(credential.credentialSubject.id)}
+            </a>
+            <span> by </span>
+            <a href="{IOTA_IDENTITY_RESOLVER}/{credential.issuer.id}" class="emphasis" target="_blank"
+                >{credential.issuer.name}
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/src/components/CredentialHeader.svelte:21]: same comment

File: identity-enabler/holder-mobile-app/src/components/CredentialHeader.svelte [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            </a>
+            <span> at </span>
+            <span class="emphasis">{new Date(credential.issuanceDate).toLocaleString()}</span>
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/src/components/CredentialHeader.svelte:24]: set a locale explicitly as in other parts of the app

File: identity-enabler/holder-mobile-app/src/lib/helpers.ts [hunk 8]
```diff
@@ -187,10 +186,6 @@ export function persistent<T>(key: string, initialValue: T, saveTransformation?:
     return state;
 }
 
-export function generateRandomId(): string {
-    return uuidv4();
-}
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/src/lib/helpers.ts:192]: why is this function removed?

File: identity-enabler/holder-mobile-app/src/components/modal/Share.svelte [hunk 9]
```diff
@@ -30,7 +30,7 @@
                 text: presentationJSON
             });
         } catch (error) {
-            console.log("Error sharing: " + error);
+            console.error("Error sharing: " + error);
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/src/components/modal/Share.svelte:33]: what happens if this happens?

File: identity-enabler/holder-mobile-app/src/pages/Credential.svelte [hunk 10]
```diff
@@ -61,19 +59,11 @@
...
             <header>
-                {#if credential.enrichment.credentialLabel === "Organisation ID"}
-                    <i class="icon-zebra credential-logo" />
-                    <p>{credential.metaInformation.issuer.toUpperCase()}</p>
-                {:else}
-                    <i class="icon-credential credential-logo" />
-                    <p>{credential.enrichment.issuerLabel}</p>
-                {/if}
-                <p>{credential.enrichment.credentialLabel}</p>
-                <p>{new Date(preparedCredentialDocument.issuanceDate).toLocaleString()}</p>
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/src/pages/Credential.svelte:72]: same comment

File: identity-enabler/holder-mobile-app/src/pages/Home.svelte [hunk 11]
```diff
@@ -176,9 +168,9 @@
                 <div transition:slide class="list">
                     <ListItem
                         icon="credential"
-                        onClick={() => navigate("credential", { state: { credential } })}
-                        heading={credential.enrichment ? credential.enrichment.issuerLabel : ""}
-                        subheading={credential.enrichment ? credential.enrichment.credentialLabel : ""}
+                        onClick={() => navigate("/credential", { state: { credential } })}
+                        heading={credential.type[1]}
+                        subheading="Issued by {credential.issuer.name}"
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/holder-mobile-app/src/pages/Home.svelte:173]: same comment

File: identity-enabler/verifier-mobile-app/rollup.config.js [hunk 12]
```diff
@@ -14,27 +14,6 @@ import * as path from "path";
...
-            server = require("child_process").spawn("npm", ["run", "start", "--", "--dev"], {
-                stdio: ["ignore", "inherit", "inherit"],
-                shell: true
-            });
-
-            process.on("SIGTERM", toExit);
-            process.on("exit", toExit);
-        }
-    };
-}
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/verifier-mobile-app/rollup.config.js:36]: same comment

File: identity-enabler/verifier-mobile-app/src/components/CredentialHeader.svelte [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    {/if}
+    {#if !hideDetails}
+        <p>
+            <span>Issued to </span>
+            <a href="{IOTA_IDENTITY_RESOLVER}/{credential.credentialSubject.id}" class="emphasis" target="_blank">
+                {shortenDID(credential.credentialSubject.id)}
+            </a>
+            <span> by </span>
+            <a href="{IOTA_IDENTITY_RESOLVER}/{credential.issuer.id}" class="emphasis" target="_blank"
+                >{credential.issuer.name}
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/verifier-mobile-app/src/components/CredentialHeader.svelte:29]: same comment

File: identity-enabler/verifier-mobile-app/src/components/CredentialHeader.svelte [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+            </a>
+            <span> at </span>
+            <span class="emphasis">{new Date(credential.issuanceDate).toLocaleString()}</span>
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/verifier-mobile-app/src/components/CredentialHeader.svelte:32]: same comment

File: identity-enabler/verifier-mobile-app/src/pages/Home.svelte [hunk 15]
```diff
@@ -127,9 +127,8 @@
                             icon={isExpired(credential.issuanceDate) ? "cross" : "check"}
                             iconColor="#1e22aa"
                             onClick={() => onClickCredential(credential)}
-                            heading={"IOTA"}
-                            subheading={credential.type[1]}
-                            expired={isExpired(credential.issuanceDate)}
+                            heading={credential.type[1]}
+                            subheading="Issued by {credential.issuer.name}"
```
Comment (jmcanterafonseca-iota) [inline identity-enabler/verifier-mobile-app/src/pages/Home.svelte:131]: same comment

**Review by jmcanterafonseca-iota — CHANGES_REQUESTED** [review]
some comments

File: identity-enabler/holder-mobile-app/src/lib/helpers.ts (hunk 8 above)
Comment (7opf) [reply identity-enabler/holder-mobile-app/src/lib/helpers.ts:192]: It's no longer used after deletions.

**Review by jmcanterafonseca-iota — APPROVED** [review]
LGTM, thanks

**Pull request merged**